### PR TITLE
Integrated HERE SDK 4.12.10 and Flutter 3.3.2

### PR DESCRIPTION
**Pull request by tstaszak89**
Removed deprecations for HERE SDK 4.12.10, deprecated MapUpdater
functions are still used (checkMapUpdate and performMapUpdate).

Signed-off-by: Tadeusz Staszak <36914750+tstaszak89@users.noreply.github.com>

### CODE REVIEW

File: lib/download_maps/map_loader_controller.dart [hunk 1]
```diff
@@ -233,99 +234,116 @@ class MapLoaderController extends ChangeNotifier {
...
   }
 
   void _onMapUpdaterComplete(MapLoaderError? error) {
     if (error != null && error != MapLoaderError.operationCancelled) {
       print("map update error ${error}");
-      _mapUpdateErrors.add(error);
+      _catalogUpdateErrors.add(error);
+    }
+    _catalogUpdateTask = null;
+    if (_catalogsToUpdate.isNotEmpty) {
```
Comment (ppiatkowski-here) [inline lib/download_maps/map_loader_controller.dart:305]: I'm not sure about this but maybe it's easier to run catalog all updates concurrently if that's supported?

We could create a completer for every catalog and initiate all updates at the same time. Then use something like this: `await Future.wait(_completerList.map((e) => e.future));` to wait until all updates are finished.

File: lib/common/util.dart [hunk 2]
```diff
@@ -289,11 +289,19 @@ Future<bool> showCommonConfirmationDialog({
 /// offline mode).
 void setTrafficLayersVisibilityOnMap(BuildContext context, HereMapController hereMapController) {
   AppPreferences appPreferences = Provider.of<AppPreferences>(context, listen: false);
-  VisibilityState trafficVisibilityState = appPreferences.useAppOffline
-      ? VisibilityState.hidden
+  bool enableTraffic = appPreferences.useAppOffline
+      ? false
       : appPreferences.showTrafficLayers
```
[the remark this answers is not in the document]
Comment (tstaszak89) [reply lib/common/util.dart]: fixed

File: lib/common/util.dart [hunk 3]
```diff
@@ -289,11 +289,19 @@ Future<bool> showCommonConfirmationDialog({
...
+  if (enableTraffic) {
+    hereMapController.mapScene.enableFeatures({
+      MapFeatures.trafficFlow: MapFeatureModes.trafficFlowWithFreeFlow,
+      MapFeatures.trafficIncidents: MapFeatureModes.trafficIncidentsAll
+    });
+  } else {
+    hereMapController.mapScene.disableFeatures([MapFeatures.trafficFlow, MapFeatures.trafficIncidents]);
+  }
+
+  MapFeatures.trafficFlow;
```
[the remark this answers is not in the document]
Comment (tstaszak89) [reply lib/common/util.dart]: fixed

File: lib/download_maps/map_loader_controller.dart (hunk 1 above)
Comment (tstaszak89) [reply lib/download_maps/map_loader_controller.dart:305]: Main problem with running all updates concurrently is to manage state of every task. It is quite hard to achieve with current UI that is designed to run one update at the time.

**Review by ppiatkowski-here — APPROVED** [review]

**Review by ppiatkowski-here — APPROVED** [review]

**Review by stefan-troyke — APPROVED** [review]

**Pull request merged**